<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.3.1/dist/css/bootstrap.min.css" integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous">
    <title>Biological Circuits</title>
  </head>
  <body>
    <br><center><h1>An Overview of Biological Circuits</h1></center><br>
    <div class="row">
      <div class="col-md-8 mx-auto p-5">
        <h2>Table of Contents:</h2>
        <ul>
          <li><a href="#Introduction">Introduction</a></li>
          <li><a href="#History">History</a></li>
          <li><a href="#Designing">Designing Biological Circuits</a></li>
          <li><a href="#Transcriptional">Transcriptional Logic Gates</a></li>
          <li><a href="#Electrical">How Biological Circuits Link With Electrical Circuits</a></li>
          <li><a href="#Importance">Importance and Future Advancements of Circuits</a></li>
          <li><a href="#References">References</a></li>
        </ul><br>

        <h2 id="Introduction">Introduction:</h2>
        <p>When we think of circuits, we usually hold a picture of bulbs and wires in our heads, but circuits can also be made with cells. Though artificial circuits made of cells were designed only in the past century, intrinsic cellular mechanisms resemble artificial logical circuits in many ways. For instance, from a macro point of view, when blood glucose levels are elevated after a meal, beta cells in the pancreas produce and secrete insulin to accelerate insulin intake by other cells, consequently restoring blood glucose to a normal level. The mechanism described above is an example of the biological circuit, and the artificial manipulation of such biological circuitry is a sub-branch of synthetic biology that focuses on performing logical operations that are previously carried out in electronic circuits. Biological circuits can be divided into two categories, molecular circuits that perform logical operations at the molecular levels in terms of DNA, RNA, and proteins, and cell circuits that mainly operate at the scale of cells. The two levels of biological circuitry are not independent, as the intrinsic cellular activity of any given cell is dependent on intracellular activities, and vice versa. Considering the example of blood glucose regulation again, the mechanism is both molecular for insulin is produced and taken by individual cells, and cellular because cell-to-cell communication is a necessary component of the regulation.</p>
        <center><figure class="figure">
          <img src="images/image1.png" class="figure-img img-fluid rounded">
          <figcaption class="figure-caption">Image credit: <a href="https://www.youtube.com/watch?v=uqOLAEmS0Mg">https://www.youtube.com/watch?v=uqOLAEmS0Mg</a></figcaption>
        </figure></center>
        <br>

        <h2 id="History">History:</h2>
        <p>One of the most studied and illustrated natural circuits is the lac operon by French biochemists Jacques Monod and Francois Jacob, who discovered the “switch” mechanism in <i>E. coli</i> that enables the bacteria to preferentially consume glucose than lactose, when the former and more digestive sugar is present <sup><a href="#R1">[1]</a></sup>. During their study of gene expression levels in <i>E. coli</i>, they discovered that five genes responsible for the production of the amino acid tryptophan are arranged in a cluster on the bacteria’s genome. This arrangement, later defined as operon, allows the clustered genes to be activated by the same signal and transcribed as a group. Monod and Jacob further developed the Jacob-Monod model in an attempt to describe the mechanism of operons, dividing the operon structure into 4 parts consisting of structural genes (genes that are later transcribed to functional protein), operator (regulatory region of the genome that serves as the binding site for repressor), promoter (regulatory region of the genome that serves as the binding site for RNA polymerase to start transcription), and repressor (a protein that inhibits transcription of the structural genes). As we will see later, the operon system is frequently employed in both natural and artificial biological circuits.</p>
        <center><figure class="figure">
          <img src="images/image2.png" class="figure-img img-fluid rounded">
          <figcaption class="figure-caption">Image credit: <a href="https://www.khanacademy.org/science/ap-biology/gene-expression-and-regulation/regulation-of-gene-expression-and-cell-specialization/a/the-lac-operon">Khan Academy</a></figcaption>
        </figure></center>
        <p>The lac operon system is an inducible system, in which the repressor is normally bonded to the operator region to inhibit the transcription of the structural genes, which in this case is lactase, the enzyme that metabolizes lactose. In order to deactivate the repressor, an inducer must connect to the repressor to break its connection from the operator region and allow efficient transcription of the lactase. The described inducible system allows <i>E. coli</i> to produce lactase only when it is needed, since the digestion of glucose (a monosaccharide, a basic form of sugar) is much more energetically efficient than the digestion of lactose (a disaccharide, which is two monosaccharides connected by glycosidic bond). Monod and Jacob later shared the 1965 Nobel Prize for their study.</p>
        <p>In comparison to the intrinsic biological circuitry in organisms that have continued to evolve for billions of years, artificial biological circuits have a much shorter but fast-developing history. In 2000, Elowitz and Leibler used three transcriptional repressor genes to build an oscillating network, termed the repressilator, that exhibits a stable oscillator with fixed oscillation periods that mimics an electrical oscillator system <sup><a href="#R2">[2]</a></sup>. The oscillation mechanism in the repressilator is achieved by a feedback loop, in which the expression of each gene activates the expression of the next gene in the loop. In the same year, Gardner and Collins designed and constructed a genetic toggle switch system consisting of two mutually repressible promoter-repressor systems <sup><a href="#R3">[3]</a></sup>. The two stable states of the toggle switch system are the low state, during which repressor 2 is transcribed and repressor 1 is repressed, and the high state, during which repressor 1 is transcribed and repressor 2 is repressed. Gardener and Collins show that the system is able to successfully switch states by change in temperature or addition of transient chemicals.</p>
        <center><figure class="figure">
          <img src="images/image3.png" class="figure-img img-fluid rounded">
          <figcaption class="figure-caption">Image credit: <a href="#R2">[2]</a></figcaption>
        </figure></center>
        <br>
        <center><figure class="figure">
          <img src="images/image4.png" class="figure-img img-fluid rounded">
          <figcaption class="figure-caption">Image credit: <a href="#R3">[3]</a></figcaption>
        </figure></center>
        <br>

        <h2 id="Designing">Designing Biological Circuits:</h2>
        <p>In this section, we will learn more about the principles of designing biological circuits from Chakraborty and Raman’s paper <i>Designing Biological Circuits: From Principles to Applications</i> <sup><a href="#R4">[4]</a></sup>. The designing aspect during biological circuit implementation is important in several ways, as properly designed biological circuits simplify the consequent implementation process, while improperly designed artificial biological circuits often deviate from expected behavior due to the fact that one biological system is often interdependent on many others and is prone to noise factors. We explore the design principles of biological circuits in four areas, namely functionality, circuit structure design, modeling, and design characteristics.</p>
        <h4>Functionality</h4>
        <p>Just as an artificial logical gate returns specific outputs in response to different inputs, biological gates are designed to produce a specific concentration profile of the protein expressed by the output gene, when the concentration profile of the protein expressed by the input gene is changed. Chakraborty and Raman describe the matching of the expressed output of the circuit in comparison to the input as the functionality of biological circuits. Going back to the examples from the introduction section, the repressible operon system takes in three repressible operon systems and produces an oscillating system of varying levels of green fluorescence. For the toggle switch system, two operon systems, combined with either heat induction or the addition of transient chemicals, enable the biological circuit to switch from one state to the other.</p>
        <h4>Circuit Structure Design</h4>
        <p>The design of genetic circuits ranges from experimental approaches such as control-theoretic approaches and rule-based design to data-centric approaches such as brute force searches and the optimization/machine learning method. As its name suggests, the brute force search performs a comprehensive search of all possible topologies to discover a module or motif of interest that could be used as biological circuits. Though studies have shown valuable insights into producing regulation networks within biological systems, it is also projected to not be computationally scalable, as the running time of the search increases exponentially with the number of nodes (DNA, RNA, enzymes, etc.) in the system. On the other hand, control-theoretic approaches adopted from the design principles of engineering systems could be used, and some of the examples adopting this approach include the in-cell feedback control that Elowitz and Leibler used in the design of the repressilator, and the toggle switch switch in <i>E. coli</i> in Gardner’s study. If extensive knowledge of a particular mechanism is already known, the rule-based design could also be used and is recently assisted by Cello, an automation software that allows users to design biological circuits with a programming knowledge. Lastly, with the growing emphasis on artificial intelligence, machine learning models could be employed to all three of the aforementioned approaches to optimize existing circuit design and discover potential targets.</p>
        <h4>Modeling</h4>
        <p>The modeling part of biological circuit design can be defined as the possible ways to theorize, measure, and represent the biochemical processes underlying the operation of the circuit. Kinetic modeling, which assumes cells’ biochemical environment to be homogeneous in space, and logical modeling, which translate gaugable biological information into logical information, are the two major types of modeling. The kinetic approach focuses on the chemical kinetic activity in cells, employing previously established biochemistry concepts such as Mass-Action Kinetics, Hill Kinetics, and Generalized Mass Action to model the designing biological circuits. Alternatively, the logical approach treats the chemical activity in cells discretely and qualitatively, regardless of the intensity of the activity. It is also noted that multiplexer, an integral part of electronic circuit design, is also within the reach of biological circuitry, as some genes express unconditionally and could consequently serve as inputs to a biological multiplexer.</p>
        <h4>Design Characteristics</h4>
        <p>We also want to mention a few characteristics often encountered during the design and implementation of biological circuits. The first is retroactivity, which is the loss of functionality of a biological circuit when it is connected with other individually functional circuits. This is often due to the interference between biological circuit signals and resources, which presents as a major drawback in comparison to electronic circuits. Particularly, within a cell hosting biological circuits, there is a limited amount of resources, such as RNA polymerase and nucleic acids, that are required for circuit output expression, and resource competition occurs when one stage of the circuit limit or even deplete the resources necessary for the other stage. Output noise is presented as another issue, but recent studies have shown that adjusting parameters within the given biological circuit is able to reduce and minimize noise to an acceptable level. </p>
        <br>

        <h2 id="Transcriptional">Transcriptional Logic Gates:</h2>
        <p>One method for creating basic circuitry in living cells is to take advantage of transcriptional interference, which changes how DNA is translated into RNA by the cell. This method can be used to control how different genes are expressed, such as by ending transcription early to suppress a gene, which allows the behavior of the cell to be changed <sup><a href="#R5">[5]</a></sup>. By carefully selecting and designing transcription factors which bind to specific sequences in the DNA, it is possible to create predictable effects which emulate the functionality of a basic logic gate, allowing the cell to respond to specific environmental factors. This makes it much more feasible to attempt to design larger or more complicated biological circuits, as it allows for a more modular design process.</p>
        <p>An example of this process can be seen by constructing a NAND gate using an AND gate and a NOT gate <sup><a href="#R6">[6]</a></sup>. The first step is to create the individual AND and NOT gates. An example construction of an AND gate is shown below.</p>
        </figure></center>
        <center><figure class="figure">
          <img src="images/image5.png" class="figure-img img-fluid rounded">
          <figcaption class="figure-caption">Image credit: <a href="#R6">[6]</a></figcaption>
        </figure></center>
        <p>Inputs are given to the gate by reacting to specific small molecules in the environment which control two promoters (P<sub>1</sub> and P<sub>2</sub>), which controls whether or not the <i>hrpR</i> and <i>hrpS</i> genes will be transcribed. The transcription of the output promoter for <i>hrpL</i> will only occur if both <i>hrpR</i> and <i>hrpS</i> genes are present, as they are required to bind the upstream activator sequence to open the <i>hrpL</i> transcription complex through ATP hydrolysis.</p>
        <p>A NOT gate can be designed in a similar fashion, using the same method for accepting input to the gate. In the example NOT gate shown below, the presence of the input gene is used along with a repressor module to invert the output.</p>
        </figure></center>
        <center><figure class="figure">
          <img src="images/image6.png" class="figure-img img-fluid rounded">
          <figcaption class="figure-caption">Image credit: <a href="#R6">[6]</a></figcaption>
        </figure></center>
        <p>These two gates can then be placed in series, creating a functional NAND gate. By combining these NAND gates in different ways, a number of combinatorial circuits can be designed.</p>
        <br>

        <h2 id="Cello">Cello — Programming the Bacterial Cell:</h2>
        <p>One particularly noteworthy advancement in the practicality of designing biological circuits came from MIT in the form of the Cello project. Cello, which was named in reference to Cellular Logic, is an open-source framework designed to automate the creation of biological circuits by providing a programming environment similar to a traditional hardware description language such as Verilog, which is used for circuit generation. In their paper, the authors describe the use of Monte Carlo simulations to select appropriate sequences of transcriptional gates whose connection is feasible, given that not all available gates can be chained together due to differences in the regulatory methods used <sup><a href="#R7">[7]</a></sup>.</p>
        <p>Cello relies on the discovery and design of modular biological logic gates similar in function to those employed by traditional circuitry to allow for the automation of genetic circuit design. This presents an interesting situation in which viable biological circuits can be created by anyone familiar with digital logic design, without necessitating advanced prior knowledge in transcriptional logic or biology in general. This makes the design of biological circuitry much more approachable to non-specialists, particularly computer scientists familiar with traditional circuitry. Although Cello is currently only designed to work in <i>E. coli</i> bacterial cells, it shows the possible potential of more general design methods in the future.</p>
        <p>To learn more about Cello, see the <a href="https://www.cidarlab.org/cello">Cidar Lab website</a> or the project’s <a href="https://github.com/CIDARLAB/cello">GitHub repository</a>.</p>
        <br>

        <h2 id="Electrical">How Biological Circuits Link With Electrical Circuits:</h2>
        <p>Everything from the very core of biology, starting at single cellular bodies, consists of molecular interactions which all are represented by intricate, large-scale networks in order to produce any type of biological circuits and systems. These networks are extremely difficult to interpret due to both the complexity of each biochemical pathway, and the links between these pathways creating interactions too compounded to compute through non-electrical solutions. So far we have dived into designing these circuits and specific advancements into helping create them as well. However, it has been proven difficult to actually attempt to design these biological circuits by linking them up with electrical circuits. One intriguing, yet viable solution brought to help decipher these circuits is examined by a paper, <i>The Merging of Biological and Electronic Circuits</i>, by Teo and Sarpeshkar which explores the usage of a four-pronged method where merging biology and electronics can help quicken analyze biological systems with room for less error. It is our hope that this information will help engineers create a blueprint or a draft representation of how to properly design these circuits <sup><a href="#R8">[8]</a></sup>.</p>
        </figure></center>
        <center><figure class="figure">
          <img src="images/image7.png" class="figure-img img-fluid rounded">
          <figcaption class="figure-caption">Image credit: <a href="#R8">[8]</a></figcaption>
        </figure></center>
        <h4>Mapping Biological Interactions to Electronic Circuits</h4>
        <p>The first step to conduct merging biological and electronic circuits is to map them out using ordinary differential equations. These equations are then illustrated at the DNA, RNA, protein, and small-molecule levels, which is given as an example of an enzyme-substrate (ES) binding reaction. In this example, we are given equations depicting the association of E and S and other mathematical data. We then picture/draw the canonical analog circuit representing the molecular biological circuits, and combining both that and the equations, ultimately provide an easier time for feedback, dynamics, robustness, and noise without mass calculations. Depending on the design used for the biological circuit, we can study its schematic equivalent using analytical tools from its design to uncover the specific properties of the circuit.</p>
        <h4>Electronic Software for Bio Circuit Design</h4>
        <p>Being able to map the circuits from equations into a blueprint design, provides an intuitive system making it easier for a designer to design, analyze, and simulate circuits and systems with many state variables, contributing the ability to switch the complexity of a model. In short, there have been many designs biologists used to port many basic electronic circuit motifs such as logic gates, latches, oscillators,etc. to biological circuits. Currently there are existing software and circuit simulation tools such as Cadence and SPICE to replicate a biological circuit to fit as an electronic circuit design.</p>
        <h4>Compiling Schematics to Hardware</h4>
        <p>While existing software and simulation tools are continuing to improve, providing the necessary information to merge biological and electrical circuits, it also requires the necessary hardware to run the simulation without any runtime errors or noticeably slow runs. Luckily in our time, large-scale bio-molecular systems can still be run at a remarkable speed  with a flat simulation time using highly parallel asynchronous analog cytomorphic systems. For biological purposes, using cytomorphic chips is deemed efficient due to its compatible readouts and programmability. It is pipelined to achieve highly stochastic simulations in the program efficiently. The only downside to these chips is a high level of model complexity. However, because biology is a subject that is moving towards a larger understanding of complex equations and systems, it requires high complexity.</p>
        <h4>Bio-Electronic Reporting Circuits Over Bio-Optical Circuits</h4>
        <p>The testing and validation process for these designs are based on making optical measurements of a reporter molecule produced by the circuit. As of now, optical biosensors are popularly used in biological circuits, however it provides heavy flaws. Some include cytotoxicity which causes reductions in range, limiting consecutive accurate measurements, and even damaging the cell itself. In order for the process to be accurate, biosensors with a wide dynamic range and good sensitivity that can provide real-time measurements are necessary. In this paper, they use the response of a biological comparator comparing a bio-optical fluorescent reporter to an electrical reporter which in conclusion, led to a more accurate measurement while providing a convenient electronic output. This proves that a safer and higher success rate is found using bio-electronic circuits over bio-optical ones.</p>
        <br>

        <h2 id="Importance">Importance and Future Advancements of Circuits:</h2>
        <p>The advancement of circuitry between biology and electricity/technology is accelerating at a rapid rate as both categories have been prominent to our society the past century or so. Engineering these circuits not only tests human limitations on changing natural biology, but can do so in a positive impact to our society. With the help of technological circuits, biomedical advancements have been and will continue to aid support for treatments such as gene therapy, enabling a way to alternate our genes so that diseases and other burdens will no longer be an issue. Some successes involve Leukemia, specific immune deficiencies, Hemophilia, and certain blindness illnesses. Synthetic circuits offer a more controlled approach to figuring out intricate biological data and can continue to provide solutions to diseases and unwanted genes that have not yet been solved yet. Just like mechanical circuitry, biological circuits have the potential to have more successful designs that we have not unlocked so far. Current measuring systems in biology are still relatively inexpensive, however they lack the accuracy, reliability, and safety of what bio-electrical circuits bring. The cost of bio-electrical circuits is relatively high, so although the methods and techniques are advancing at a fast rate, we currently do not own the materials to catch up with that speed. Because of the complex infrastructural networks biological information contains, it will take time before it can comfortably solve most of the world’s biological problems we have today. But hopefully with the continuation of technological advancements, we could intelligently use biological circuitry with computer science to maximize the potential of gene and cell editing.</p>
        <br>

        <h2 id="References">References:</h2>
        <ol>
          <li id="R1" style="margin-bottom: 1em">Griffiths, Anthony J.F.; Wessler, Susan R.; Carroll, Sean B.; Doebley, John (2015). <i>An Introduction to Genetic Analysis</i> (11 ed.). Freeman, W.H. & Company. pp. 400–412.</li>
          <li id="R2" style="margin-bottom: 1em">Elowitz MB, Leibler S. <i>A synthetic oscillatory network of transcriptional regulators</i>. Nature. 2000 Jan 20;403(6767):335-8. doi: 10.1038/35002125. PMID: 10659856.</li>
          <li id="R3" style="margin-bottom: 1em">Gardner TS, Cantor CR, Collins JJ. <i>Construction of a genetic toggle switch in Escherichia coli</i>. Nature. 2000 Jan 20;403(6767):339-42. doi: 10.1038/35002131. PMID: 10659857.</li>
          <li id="R4" style="margin-bottom: 1em">Chakraborty, Debomita & Rengaswamy, Raghunathan & Raman, Karthik. (2021). <i>Designing biological circuits: from principles to applications</i>.</li>
          <li id="R5" style="margin-bottom: 1em">Escalas Bordoy, Antoni & O'Connor, Nolan & Chatterjee, Anushree. (2019). <i>Construction of Two-Input Logic Gates Using Transcriptional Interference</i>. ACS Synthetic Biology. 2019. 10.1021/acssynbio.9b00321. </li>
          <li id="R6" style="margin-bottom: 1em">Wang, B., Kitney, R., Joly, N. et al. <i>Engineering modular and orthogonal genetic logic gates for robust digital-like synthetic biology</i>. Nat Commun 2, 508 (2011). doi: 10.1038/ncomms1516</li>
          <li id="R7" style="margin-bottom: 1em">Nielsen AA, Der BS, Shin J, Vaidyanathan P, Paralanov V, Strychalski EA, Ross D, Densmore D, Voigt CA. <i>Genetic circuit design automation</i>. Science. 2016 Apr 1;352(6281):aac7341. doi: 10.1126/science.aac7341. PMID: 27034378.</li>
          <li id="R8">Jonathan J.Y. Teo, Rahul Sarpeshkar. <i>The Merging of Biological and Electronic Circuits</i>. iScience, Volume 23, Issue 11, 2020, 101688, ISSN 2589-0042. doi : 10.1016/j.isci.2020.101688.</li>
        </ol><br>
      </div>
    </div>
</body>
</html>
